db/compaction/compaction.cc

```diff
@@ -426,33 +467,39 @@ bool Compaction::OverlapPenultimateLevelOutputRange(
   if (!SupportsPerKeyPlacement()) {
     return false;
   }
+
+  // See FIXME in Compaction::PopulatePenultimateLevelOutputRange().
+  // We do not compact any key up in this case.
+  if (penultimate_level_smallest_.size() == 0 ||
+      penultimate_level_largest_.size() == 0) {
+    return false;
+  }
+
   const Comparator* ucmp =
       input_vstorage_->InternalComparator()->user_comparator();
 
   return ucmp->CompareWithoutTimestamp(
-             smallest_key, penultimate_level_largest_user_key_) <= 0 &&
+             smallest_key, penultimate_level_largest_.user_key()) <= 0 &&
          ucmp->CompareWithoutTimestamp(
-             largest_key, penultimate_level_smallest_user_key_) >= 0;
+             largest_key, penultimate_level_smallest_.user_key()) >= 0;
```

Comment:
If it were me I'd probably just make the comparison here and in `WithinPenultimateLevelOutputRange()` exclusive. But your ways nicer I think.

Reply:
Thanks, I did not think of that way, it should work too.